manifests/logical_volume.pp

```diff
@@ -68,18 +85,38 @@
   }
 
   if $createfs or $ensure != 'present' {
-    exec { "ensure mountpoint '${mountpath}' exists":
-      path    => [ '/bin', '/usr/bin' ],
-      command => "mkdir -p ${mountpath}",
-      unless  => "test -d ${mountpath}",
-    } ->
-    mount { $mountpath:
+    if $fs_type == 'swap' {
+      if $ensure == 'present' {
+        exec { "swapon for '${mount_title}'":
+          path      => [ '/bin', '/usr/bin', '/sbin' ],
+          command   => "swapon ${lvm_device_path}",
+          unless    => "grep `readlink -f ${lvm_device_path}` /proc/swaps",
```

Comment:
Hm, sorry, but that doesn't really work in my case. It looks like this:
* ${lvm_device_path} = ${mount_title} = /dev/vg0/swap, which is a link to /dev/dm-2
* The output of `swapon -s` lists /dev/mapper/vg0-swap, which is also a link to /dev/dm-2
* The output of `cat /proc/swaps` lists /dev/dm-2

So in order to use the output of `swapon -s` we either have to extract the device and follow its redirect with `readlink -f` and do the same for ${lvm_device_path} to make the comparison work, or we adjust the way the device path is built. At the moment ${lvm_device_path} reflects the logic, which was used up until now and sets the path to /dev/${volume_group}/${name}. Instead of that we could probably use /dev/mapper/${volume_group}-${name}. That would make the comparison of the output of `swapon -s` easier.

Is there any specific problem with my approach, which doesn't occur with your proposed alternative?